### ISSUE TITLE
NVDA cannot announce candidate item in default Chinese Input method Microsoft pinyin

### ISSUE THREAD

**Issue by jiangtiandao** [nvaccess/nvda#8562]
### Steps to reproduce:
Install  Chinese Language pack
Reboot to load language pack 
Open Notepad
Press Windows + Space  switch Input Method to Chinese Simplified China Microsoft pinyin
press h and e
then press down arrow several times
### Actual behavior:
NVDA keeps speaking he 
### Expectedhavior: 
Candidate item is announced 
If you use english espeak it would be "Chinese letter"
#### NVDA Installed/portable/running from source:
installed
#### NVDA version:
2018.2.1
#### Windows version:
Windows 8.1
#### Name and version of other software in use when reproducing the issue:
Windows 10 and 7 also has this issue 
### Other questions:
Narrator can read this.
NVDA uses ITfUIElementSink to get candidate item
 Is it possible to get candidate item from UIAutomation for Microsoft pinyin?
#### Does the issue still occur after restarting your PC?
Yes
#### Have you tried any other versions of NVDA?
alpha-15682,48201d01
This version also has this issue

**Comment by dingpengyu** [comment]
Hello, I am using the NVDA alpha15690 development board to build, I also encountered the same problem when using Microsoft Pinyin to input Chinese.

**Comment by jiangtiandao** [comment]
I found some information about the candidate item UI through Inspect.exe in windows SDK.
Name:	"Modern Candidate UI"
ControlType:	UIA_PaneControlTypeId (0xC371)
LocalizedControlType:	"pane"
BoundingRectangle:	{l:168 t:132 r:749 b:180}
IsEnabled:	true
IsOffscreen:	false
IsKeyboardFocusable:	false
HasKeyboardFocus:	false
ProcessId:	15732
RuntimeId:	[2A.60E58]
FrameworkId:	"DirectUI"
ClassName:	"CCandidateViewHWNDElement"

**Comment by man0528** [comment]
I'm very upset with this question
I was very upset about this problem and I didn't choose it.

**Comment by man0528** [comment]
I'm very upset about this problem

**Comment by larry801** [comment]
I cherry picked this commit into master branch. 
https://github.com/nvaccess/nvda/commit/dbbaacea10ac1ec8c07dbe943535d86284546861
Most Chinese character input bug is resolved.
It seems that Microsoft has fixed UIA events issue existed in 2015.
I tested in Win8.1 x64 and Win10 17134.191

@michaelDCurran 

That commit will also fix the following issue.
#8343 
#4011
#3983
